### PR TITLE
K8s remove exc svc

### PR DESCRIPTION
**Pull request by godber**
This PR does the following

* removes the Kubernetes Service that sat between the workers and execution controller pods.  The kubernetes deployment for the workers won't even get created until after the execution controller pod is in the `Running` state and the `podIp` is available. refs: #1009, #1649 
* removes the call to `waitForTcpOpen` in the messaging library. refs: #1649

### CODE REVIEW

**Comment by godber** [comment]
@peterdemartini ... I should probably remove `waitForTcpOpen` huh?  I've removed the call from the messaging module (there not much point now).

I should remove the util function and tests too right?
https://github.com/terascope/teraslice/blob/master/packages/teraslice-messaging/src/utils/index.ts#L41

**Comment by peterdemartini** [comment]
> @peterdemartini ... I should probably remove `waitForTcpOpen` huh?  I've removed the call from the messaging module (there not much point now).
> 
> 
> 
> I should remove the util function and tests too right?
> 
> https://github.com/terascope/teraslice/blob/master/packages/teraslice-messaging/src/utils/index.ts#L41

Sure, since it didn’t help, we can remove it, if we ever need it back we can look it up in git history.

File: packages/teraslice/lib/cluster/services/cluster/backends/kubernetes/k8s.js [hunk 1]
```diff
@@ -57,6 +59,38 @@ class K8s {
...
+     *
+     * NOTE: If your selector will return multiple pods, this method probably
+     * won't work for you.
+     * @param {String} selector kubernetes selector, like 'controller-uid=XXX'
+     * @param {String} ns       namespace to search, this will override the default
+     * @param {Number} timeout  time, in ms, to wait for pod to start
+     */
+    async waitForSelectedPod(selector, ns, timeout = 10000) {
+        const namespace = ns || this.defaultNamespace;
+        let now = (new Date()).getTime();
```
Comment (peterdemartini) [inline packages/teraslice/lib/cluster/services/cluster/backends/kubernetes/k8s.js:74]: I prefer `Date.now()` for simplicity

**Comment by godber** [comment]
Assuming these test pass (they should) ... this is ready to go.

File: packages/teraslice/lib/cluster/services/cluster/backends/kubernetes/k8s.js [hunk 2]
```diff
@@ -57,6 +59,38 @@ class K8s {
...
+        // eslint-disable-next-line no-constant-condition
+        while (true) {
+            const result = await this.client.api.v1.namespaces(namespace)
+                .pods().get({ qs: { labelSelector: selector } });
+
+            // NOTE: This assumes the first pod returned.
+            const pod = result.body.items[0];
+
+            if (pod.status.phase === 'Running') return pod;
+            if (now > end) throw new Error('timeout');
```
Comment (peterdemartini) [inline packages/teraslice/lib/cluster/services/cluster/backends/kubernetes/k8s.js:86]: This error message should be more descriptive

File: packages/teraslice/lib/cluster/services/cluster/backends/kubernetes/k8s.js [hunk 3]
```diff
@@ -57,6 +59,38 @@ class K8s {
...
+            this.logger.debug(`waiting for pod matching: ${selector}`);
+
+            await pDelay(500);
+            now = (new Date()).getTime();
```
Comment (peterdemartini) [inline packages/teraslice/lib/cluster/services/cluster/backends/kubernetes/k8s.js:90]: Use `Date.now()`

Comment (godber) [reply packages/teraslice/lib/cluster/services/cluster/backends/kubernetes/k8s.js]: Geez there was a second one of those?

**Comment by godber** [comment]
Hold on ... I accidentally committed some garbage to example jobs.  Don't merge.

**Pull request merged**